source/extensions/tracers/opentelemetry/resource_detectors/environment/environment_resource_detector.h

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include "envoy/server/factory_context.h"
+
+#include "source/common/common/logger.h"
+#include "source/extensions/tracers/opentelemetry/resource_detectors/resource_detector.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Tracers {
+namespace OpenTelemetry {
+
+/**
+ * @brief A resource detector that extracts attributes from the OTEL_RESOURCE_ATTRIBUTES environment
+ * variable.
+ * @see
+ * https://github.com/open-telemetry/opentelemetry-specification/blob/v1.24.0/specification/resource/sdk.md#detecting-resource-information-from-the-environment
+ *
+ */
+class EnvironmentResourceDetector : public ResourceDetector, Logger::Loggable<Logger::Id::tracing> {
+public:
+  EnvironmentResourceDetector(Server::Configuration::TracerFactoryContext& context)
+      : context_(context) {}
+  Resource detect() override;
+
+private:
+  Server::Configuration::TracerFactoryContext&
+      context_; // TODO, is keeping a reference ok (ownership)?
```

Comment:
I was not sure if it's OK to keep this reference in the detector. The detector needs it to access the `api()` method, which is then used on the `DataSource` to read the env variable. It will also be needed in other detectors, that read from a file for example.